fuel-vm/src/interpreter/metadata.rs

```diff
@@ -489,14 +489,14 @@ impl<Tx> GTFInput<'_, Tx> {
             GTFArgs::OutputCoinAmount => tx
                 .outputs()
                 .get(b)
-                .filter(|o| o.is_coin())
+                .filter(|o| o.is_coin() || o.is_variable() || o.is_change())
```

Comment:
Since the change amount isn't set until post-execution, this will always be zero for change outputs.

Reply:
True. Do you think we should panic here rather than returning the zero? The old error is `OutputNotFound` which seems a bit misleading. So is returning zero.

Reply:
Reverted to old behavior in f7855c9.